node/src/actors/chain_manager/mining.rs

```diff
@@ -246,8 +247,15 @@ impl ChainManager {
             .wait(ctx);
     }
 
+    /// Check whether the reward covers the minimum specified collateral
+    pub fn check_reward_to_collateral_filter(&self, dr_state: &DataRequestState) -> bool {
+        let dr_reward_to_collateral_ratio =
+            (dr_state.data_request.witness_reward / dr_state.data_request.collateral) * 1000;
```

Comment:
You need to use floats here, otherwise `witness_reward / collateral` will return `0` if `witness_reward < collateral`
```suggestion
            (f32::from(dr_state.data_request.witness_reward) / f32::from(dr_state.data_request.collateral)) * 1000.0;
```